core/geometry/internaldocs/Spiral.md

```diff
@@ -0,0 +1,133 @@
+# Spirals
+
+## General Math
+
+A spiral is a curve that winds around a point (usually the origin), moving progressively closer or farther away from it.
+
+Mathematically, a spiral can be described in polar coordinates $(r,θ)$ where $r$ is the distance from the origin and $θ$ is the angle (in radians) from the x-axis. Therefore, a spiral can be defined by $r=f(θ)$ where $f(θ)$ describes how the radius grows as the angle increases.
+
+**Example:**
+
+*Archimedean Spiral:*
+$$r=a+bθ$$
+Parametric (Cartesian) form:
+$$x(θ)=(a+bθ)cosθ \\ y(θ)=(a+bθ)sinθ​$$
+
+*Logarithmic Spiral:*
+$$r=ae^{bθ}$$
+Parametric (Cartesian) form:
+$$x(θ)=ae^{bθ}cosθ \\ y(θ)=ae^{bθ}sinθ​$$
+
+**Note:** core-geometry does not currently support Archimedean or logarithmic spirals, and there are no plans to implement them. Their inclusion here is solely to illustrate the mathematical concepts behind spirals.
+-----------
+
+Another way to represent a spiral is via its curvature. In general, for a curve parameterized by curve length $\bar{s}$ we have:
+
+$$\frac{dθ}{d\bar{s}} = κ(\bar{s})$$
+
+where where $κ(\bar{s})$ is curvature. If $p$ is the position vector we have
+
+$$p(s) = (x'(\bar{s}),y'(\bar{s})) = (cosθ(\bar{s}), sinθ(\bar{s}))$$
+
+so integrating gives positions:
+
+$$x(\bar{s})= \int_{0}^{\bar{s}} cosθ(u)du \\ y(\bar{s})= \int_{0}^{\bar{s}} sinθ(u)du $$
+
+For a spiral defined by curvature varying smoothly with length, we have fraction $s = \frac{\bar{s}}{L}$ or $\bar{s} = Ls$ where $L$ is the total spiral length. This gives:
+
+$$θ(s)= \int_{0}^{Ls} κ(u)du$$
+
+Replacing this into previous formula and switching variable to fraction form gives:
+
+$$\boxed{
+\begin{aligned}
+  x(s)= L \int_{0}^{s} \Bigl(\int_{0}^{Lu} 𝜅(v)dv\Bigl) du \\ y(s)= L \int_{0}^{s} \Bigl(\int_{0}^{Lu} 𝜅(v)dv\Bigl) du
+\end{aligned}
+}$$
+
+**Example:**
+
+*Clothoid (Euler Spiral)*
+
+Curvature $𝜅$ increases linearly with arc length $s$:
+$$𝜅(s) = cs \Rightarrow θ(s) = \frac{cs^2}{2} $$
+where $c$ is a constant.
+
+Parametric (Cartesian) form:
+$$x(s)= \int_{0}^{s} cos\Bigl(\frac{cu^2}{2}\Bigl)du \\ y(s)= \int_{0}^{s} sin\Bigl(\frac{cu^2}{2}\Bigl)du $$
+
+![>](./figs/Spiral/spirals.png)
+
+## Code
+
+**TransitionSpiral3d**
+
+Parent class for all iTwin spirals is `TransitionSpiral3d` which is extended by `IntegratedSpiral3d` and `DirectSpiral3d`.
+
+`TransitionSpiral3d` class has an member called `TransitionConditionalProperties` which encapsulates 5 elements: `radius0`, `radius1`, `bearing0`, `bearing1`, and `curveLength`.
+
+We already know:
+$$θ(s)= \int_{0}^{Ls} 𝜅(u)du$$
+This can be rewritten as
+$$θ_1 - θ_0 = L \bar{𝜅} = L \frac{𝜅_0 + 𝜅_1}{2}$$
```

Comment:
I think this only applies to clothoids, not all TransitionSpirals.

Reply:
Indeed average function is not just average of start and end values. However, `TransitionSpiral3d` has a member `_designProperties` of type `TransitionConditionalProperties` and if you look at the doc for that class, you see this equation. This class does not mention clothoids. 

For example, look at `IntegratedSpiral3d.createFrom4OutOf5` which instantiate `TransitionConditionalProperties` and takes any `spiralType`.

Reply:
Ah. I think I understand now. TransitionConditionalProperties is not instantiated for DirectSpiral3ds, but only for IntegratedSpiral3ds, which is a class that generalizes the clothoid. An IntegratedSpiral3d allows the curve's curvature function to be nonlinear while still satisfying the same avg value formula enjoyed by the clothoid with its linear curvature. The NormalizedTransition class defines the 3 constraints on the normalized form of a curvature function to guarantee this avg value formula holds. So my original comment still stands: the average value formula does not hold for all TransitionSpiral3ds.

Reply:
ok I will modify the md doc to reflect this.